tests/PactNet.Native.Tests/NativePactBuilderTests.cs

```diff
@@ -39,23 +56,147 @@ public void UponReceiving_WhenCalled_CreatesNewInteraction()
                 .Throws<InvalidOperationException>();
 
-            Action action = () => this.builder.Build();
+            Action action = () => this.builder.Verify(Success);
 
             action.Should().Throw<InvalidOperationException>("because the mock server failed to start");
         }
+
+        [Fact(Skip = "Enable this when the released FFI library supports getting mock server logs")]
+        public void Verify_NoLogs_DoesNotWriteLogs()
+        {
+            this.builder.Verify(Success);
+
+            this.mockOutput.Verify(o => o.WriteLine(It.IsAny<string>()), Times.Never);
+        }
+
+        [Fact(Skip = "Enable this when the released FFI library supports getting mock server logs")]
+        public void Verify_Logs_WritesLogsToOutput()
+        {
+            this.mockServer.Setup(s => s.MockServerLogs(this.serverUri.Port)).Returns("some logs");
+
+            this.builder.Verify(Success);
+
+            this.mockOutput.Verify(o => o.WriteLine("some logs"));
+        }
+
+        [Fact]
+        public void Verify_InteractionThrowsException_RethrowsInteractionException()
+        {
+            this.builder
+                .Invoking(b => b.Verify(Error))
+                .Should().ThrowExactly<Exception>("because the inner interaction call failed");
+        }
+
+        [Fact]
+        public void Verify_NoMismatches_WritesPactFile()
+        {
+            this.builder.Verify(Success);
+
+            this.mockServer.Verify(s => s.WritePactFile(this.serverUri.Port, this.config.PactDir, false));
+        }
+
+        [Fact]
+        public void Verify_NoMismatches_ShutsDownMockServer()
+        {
+            this.builder.Verify(Success);
+
+            this.mockServer.Verify(s => s.CleanupMockServer(this.serverUri.Port));
+        }
+
+        [Fact]
+        public void Verify_FailedToWritePactFile_ThrowsInvalidOperationException()
+        {
+            this.mockServer
+                .Setup(s => s.WritePactFile(It.IsAny<int>(), It.IsAny<string>(), It.IsAny<bool>()))
+                .Throws<InvalidOperationException>();
+
+            Action action = () => this.builder.Verify(Success);
+
+            action.Should().Throw<InvalidOperationException>();
+        }
+
+        [Fact]
+        public void Verify_Mismatches_DoesNotWritePactFile()
+        {
+            this.mockServer.Setup(s => s.MockServerMismatches(this.serverUri.Port)).Returns("some mismatches");
+
+            try
+            {
+                this.builder.Verify(Success);
+            }
+            // ReSharper disable once EmptyGeneralCatchClause
+            catch
+            {
+            }
+
+            this.mockServer.Verify(s => s.WritePactFile(It.IsAny<int>(), It.IsAny<string>(), It.IsAny<bool>()), Times.Never);
+        }
+
+        [Fact]
+        public void Verify_Mismatches_WritesMismatchesToOutput()
+        {
+            this.mockServer.Setup(s => s.MockServerMismatches(this.serverUri.Port)).Returns("some mismatches");
```

Comment:
same thing, I would have a variable expectedLogs I'd use line 152 and 163

Reply:
Introduced local constant for this